### PR TITLE
IssueLabelToken with variants

### PR DESCRIPTION
**Pull request by lukasoppermann**
This PR updates the `IssueLabelToken` component to use `variants` for color instead of hex colors.
Hex colors are still supported to provide backwards compatibility.

I added a new `IssueLabelToken` component to the draft folder.

### Rollout strategy

- [ ] Patch release
- [ ] Minor release
- [ ] Major release; if selected, include a written rollout or migration plan
- [ ] None; if selected, include a brief description as to why

### Merge checklist

- [ ] Added/updated tests
- [ ] Added/updated documentation
- [ ] Added/updated previews (Storybook)
- [ ] Changes are [SSR compatible](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#ssr-compatibility)
- [ ] Tested in Chrome
- [ ] Tested in Firefox
- [ ] Tested in Safari
- [ ] Tested in Edge
- [ ] (GitHub staff only) Integration tests pass at github/github ([Learn more about how to run integration tests](https://github.com/github/primer-engineering/blob/main/how-we-work/testing-primer-react-pr-at-dotcom.md))

### CODE REVIEW

File: packages/react/src/drafts/IssueLabelToken/getColorsFromHex.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  if (h >= 58 && h <= 316 && s > 80) {
+    s = 80
+  }
+  const textColor = getColorWithContrast(
+    hsluvToHex({h, s, l: 50}),
+    backgroundColor,
+    ratio,
+    lightnessIncrement as 1 | -1,
+  )
+  // return
```
Comment (joshblack) [inline packages/react/src/drafts/IssueLabelToken/getColorsFromHex.ts:38]:
```suggestion
```

File: packages/react/src/drafts/IssueLabelToken/getColorsFromHex.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
+import {getContrast} from 'color2k'
+import {Hsluv} from 'hsluv'
+
+export const getColorsFromHex = (colorHex: string, colorScheme = 'light', isSelected = false, bgColor: string) => {
```
Comment (joshblack) [inline packages/react/src/drafts/IssueLabelToken/getColorsFromHex.ts:4]: Could you add more details or a summary to the code here? It seems like it's doing a lot of important stuff but I don't necessarily know what each of the numbers are for so any additional info in the source would be very appreciated 😅

**Review by joshblack — COMMENTED** [review]
Just wanted to leave a quick round of comments after today! I feel like something worth investigating as next steps is if we can roll this out in the existing component or not. Let me know if I can help out with that at all!

File: packages/react/src/drafts/IssueLabelToken/IssueLabelToken.hex.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import IssueLabelToken from './IssueLabelToken'
+import Box from '../../Box'
+import {hexString} from '../utils/isHex'
+
+export default {
+  title: 'Drafts/Components/IssueLabelToken/Hex',
+  component: IssueLabelToken,
+}
+
+export const Hex = ({hex, text, interactive, ...args}: {hex: hexString; interactive: boolean; text: string}) => {
```
[the remark this answers is not in the document]
Comment (lukasoppermann) [reply packages/react/src/drafts/IssueLabelToken/IssueLabelToken.hex.stories.tsx:11]: I am not really sure of the conventions we have here. I guess we could group interactive and hex into a features story.

I there some convention around this?

**Comment by lukasoppermann** [comment]
> Just wanted to leave a quick round of comments after today! I feel like something worth investigating as next steps is if we can roll this out in the existing component or not. Let me know if I can help out with that at all!

I'd love to get your help with this.

But one thing we need to keep in mind this PR changes two major things:
1. Adds `variant` property (not breaking)
2. Change the color output for any given `fillColor` provided (this is at least visually very different)

File: packages/react/src/drafts/IssueLabelToken/IssueLabelToken.hex.stories.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (joshblack) [reply packages/react/src/drafts/IssueLabelToken/IssueLabelToken.hex.stories.tsx]: @lukasoppermann for conventions, right now we have the default and playground stories in `ComponentName.stories.tsx` and then features in `ComponentName.features.tsx`. Sometimes there are examples in `ComponentName.examples.tsx` but that isn't used as much as the former two.

Hope that makes sense, just wanted to share the types of stories that we have currently. Let me know if you have any questions or need clarification (or if this doesn't really fit into any of these types of stories)

**Pull request closed without merging**